src/components/Editor/Attachments/AttachmentsList.vue

```diff
@@ -0,0 +1,241 @@
+<template>
+	<div id="attachments">
+		<input ref="localAttachments"
+			class="attachments-input"
+			type="file"
+			multiple
+			@change="onLocalAttachmentSelected">
+		<div class="attachments-summary">
+			<div class="attachments-summary-inner">
+				<Paperclip :size="20" />
+				<div v-if="attachments.length > 0">
+					{{ n('calendar', '{count} attachment', '{count} attachments', attachments.length, { count: attachments.length }) }}
+				</div>
+				<div v-else>
+					{{ t('calendar', 'No attachments') }}
+				</div>
+			</div>
+
+			<Actions>
+				<template #icon>
+					<Plus :size="20" />
+				</template>
+				<ActionButton @click="openFilesModal()">
+					<template #icon>
+						<Folder :size="20" />
+					</template>
+					{{ t('calendar', 'Add from Files') }}
+				</ActionButton>
+				<ActionButton @click="clickOnUploadButton">
+					<template #icon>
+						<Upload :size="20" />
+					</template>
+					{{ t('calendar', 'Upload from device') }}
+				</ActionButton>
+			</Actions>
+		</div>
+		<div v-if="attachments.length > 0">
+			<ul class="attachments-list-item">
+				<ListItem v-for="attachment in attachments"
+					:key="attachment.path"
+					:force-display-actions="true"
+					:title="getBaseName(attachment.fileName)"
+					@click="openFile(attachment.uri)">
+					<template #icon>
+						<img :src="getPreview(attachment)" class="attachment-icon">
+					</template>
+					<template #actions>
+						<ActionButton @click="deleteAttachmentFromEvent(attachment)">
+							<template #icon>
+								<Close :size="20" />
+							</template>
+							{{ t('calendar', 'Delete file') }}
+						</ActionButton>
+					</template>
+				</ListItem>
+			</ul>
+		</div>
+	</div>
+</template>
+
+<script>
+import ListItem from '@nextcloud/vue/dist/Components/NcListItem'
```

Comment:
i think the imports are probably missing the file endings...

Reply:
Done via [e532920](https://github.com/nextcloud/calendar/pull/4251/commits/e532920b514c965edbd96e6138dafbaeba82eb52)

`150 problems (0 errors, 150 warnings)` =)